### ISSUE TITLE
"Error in Sys.setenv(PATH = new_path) : wrong length for argument"

### ISSUE THREAD

**Issue by Stephen-G** [rstudio/reticulate#1155]
Hello, 

I have been using v1.22 without any issues and just upgraded to v1.24. Now when I try to connect I get the error message:
```
"Error in Sys.setenv(PATH = new_path) : wrong length for argument"

> library(reticulate)
#This is the command that worked in v1.22
> use_condaenv("wfs", required = TRUE)
Error in Sys.setenv(PATH = new_path) : wrong length for argument

> conda_list()
  name                              python
1 base            C:\\Anaconda3/python.exe
2  wfs C:\\Anaconda3\\envs\\wfs/python.exe

> py_config()
Error in Sys.setenv(PATH = new_path) : wrong length for argument

> use_python("C:\\Anaconda3\\envs\\wfs\\python.exe", required = TRUE)
Error in Sys.setenv(PATH = new_path) : wrong length for argument

> utils::sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    
system code page: 65001

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.24

loaded via a namespace (and not attached):
 [1] compiler_4.1.2  here_1.0.1      Matrix_1.4-0    rprojroot_2.0.2 tools_4.1.2     rappdirs_0.3.3 
 [7] Rcpp_1.0.8      grid_4.1.2      knitr_1.37      jsonlite_1.7.3  xfun_0.29       png_0.1-7      
[13] lattice_0.20-45

> conda_list()
  name                              python
1 base            C:\\Anaconda3/python.exe
2  wfs C:\\Anaconda3\\envs\\wfs/python.exe

> use_condaenv("wfs", required = TRUE)
Error in Sys.setenv(PATH = new_path) : wrong length for argument

> Sys.getenv("PATH")
[1] "C:\\rtools40\\usr\\bin;C:\\Program Files\\R\\R-4.1.2\\bin\\x64;C:\\Anaconda3;C:\\Anaconda3\\Library\\mingw-w64\\bin;C:\\Anaconda3\\Library\\usr\\bin;C:\\Anaconda3\\Library\\bin;C:\\Anaconda3\\Scripts;C:\\oracle\\product\\11.2.0\\client_1\\bin;C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files (x86)\\Microsoft SQL Server\\Client SDK\\ODBC\\130\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\DTS\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\ManagementStudio\\;C:\\Program Files\\Git\\cmd;C:\\Users\\stephen.grant\\AppData\\Local\\Microsoft\\WindowsApps;"
```
I found this tix https://github.com/conda/conda/issues/10690 but since it was previously working I'm not sure it's the same issue.

Any help would be appreciated.

**Comment by Stephen-G** [comment]
Update:
I downgraded to v1.22 and it is working again.

**Comment by t-kalinowski** [comment]
Hi, thank you for filing. Can you tell me the output the output from:
```R
python <- "C:\\Anaconda3\\envs\\wfs/python.exe"
conda_info <- reticulate:::get_python_conda_info(python)
new_path <- reticulate:::conda_run(
  "python",
  c("-c", shQuote("import os; print(os.environ['PATH'])")),
  conda = conda_info$conda,
  envname = conda_info$root,
  stdout = TRUE
)
new_path
```

**Comment by Stephen-G** [comment]
Sure. Run from new session.

> python <- "C:\\Anaconda3\\envs\\wfs/python.exe"
> conda_info <- reticulate:::get_python_conda_info(python)
Error: object 'get_python_conda_info' not found
> new_path <- reticulate:::conda_run(
+     "python",
+     c("-c", shQuote("import os; print(os.environ['PATH'])")),
+     conda = conda_info$conda,
+     envname = conda_info$root,
+     stdout = TRUE
+ )
Error: object 'conda_run' not found
> new_path
Error: object 'new_path' not found

**Comment by t-kalinowski** [comment]
You'll have to upgrade to the dev or release version of reticulate for that snippet to work.

**Comment by Stephen-G** [comment]
[1] "This program is blocked by group policy. For more information, contact your system administrator."                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
[2] "C:\\ANACON~1;C:\\ANACON~1\\Library\\mingw-w64\\bin;C:\\ANACON~1\\Library\\usr\\bin;C:\\ANACON~1\\Library\\bin;C:\\ANACON~1\\Scripts;C:\\ANACON~1\\bin;C:\\rtools40\\usr\\bin;C:\\Program Files\\R\\R-4.1.2\\bin\\x64;C:\\Anaconda3;C:\\Anaconda3\\Library\\mingw-w64\\bin;C:\\Anaconda3\\Library\\usr\\bin;C:\\Anaconda3\\Library\\bin;C:\\Anaconda3\\Scripts;C:\\oracle\\product\\11.2.0\\client_1\\bin;C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files (x86)\\Microsoft SQL Server\\Client SDK\\ODBC\\130\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\DTS\\Binn\\;C:\\Program Files (x86)\\Microsoft SQL Server\\140\\Tools\\Binn\\ManagementStudio\\;C:\\Program Files\\Git\\cmd;C:\\Users\\stephen.grant\\AppData\\Local\\Microsoft\\WindowsApps;"

**Comment by t-kalinowski** [comment]
Thank you for reporting! I just merged a PR, this should be fixed now. Please try with the development version of reticulate and let me know if you still run into this issue.
```r
remotes::install_github("rstudio/reticulate")
```

**Comment by Stephen-G** [comment]
Thanks for helping! Installed as above. Reticulate now loads without error but not able to use functions in package.

> use_condaenv("wfs", required = TRUE)
Warning in system(cmd, intern = intern, wait = wait | intern, show.output.on.console = wait,  :
  running command 'C:\WINDOWS\system32\cmd.exe /c C:\Users\STEPHE~1.GRA\AppData\Local\Temp\Rtmpiq1eb4\file33ac7e957ddf.bat' had status 1

Also if it helps:

> reticulate::repl_python()
Warning in system(cmd, intern = intern, wait = wait | intern, show.output.on.console = wait,  :
  running command 'C:\WINDOWS\system32\cmd.exe /c C:\Users\STEPHE~1.GRA\AppData\Local\Temp\Rtmpiq1eb4\file33ac75491733.bat' had status 1
Warning in system(cmd, intern = intern, wait = wait | intern, show.output.on.console = wait,  :
  running command 'C:\WINDOWS\system32\cmd.exe /c C:\Users\STEPHE~1.GRA\AppData\Local\Temp\Rtmpiq1eb4\file33ac5523aa9.bat' had status 1
>>> 
>>> import pandas as pd
C:\Program Files\R\R-4.1.2\library\reticulate\python\rpytools\loader.py:39: UserWarning: mkl-service package failed to import, therefore Intel(R) MKL initialization ensuring its correct out-of-the box operation under condition when Gnu OpenMP had already been loaded by Python process is not assured. Please install mkl-service package, see http://github.com/IntelPython/mkl-service

**Comment by t-kalinowski** [comment]
Maybe you don't have permissions to execute *.bat scripts on windows?
Can you step through `reticulate:::conda_run2_windows` and/or run this snippet and tell me the output?

```R
python <- "C:\\Anaconda3\\envs\\wfs/python.exe"
conda_info <- reticulate:::get_python_conda_info(python)
reticulate:::conda_run2(
  "python",
  c("-c", shQuote("import os; print(os.environ['PATH'])")),
  conda = conda_info$conda,
  envname = conda_info$root,
  intern = FALSE
)
```

**Comment by Stephen-G** [comment]
Yes. I agree, it looks like it's a permissions issue. Do you know specifically what command is being executed in v1.24 that wasn't in v1.22 so I can get it whitelisted? I'm no expert but it looks like the .bat is in a temp folder that could potentially change with every new session? I might have a hard time getting IT to do a blanket call to every .bat (temp folder is probably even more specious) vs a specific file.

[1] "This program is blocked by group policy. For more information, contact your system administrator."
attr(,"status")
[1] 1
Warning message:
In system(cmd, intern = intern, wait = wait | intern, show.output.on.console = wait,  :
  running command 'C:\WINDOWS\system32\cmd.exe /c C:\Users\STEPHE~1.GRA\AppData\Local\Temp\RtmpuYRkUZ\file3acc3aa84e6e.bat' had status 1

**Comment by t-kalinowski** [comment]
When a conda python installation is being initialized we need to temporarily modify the PATH variable. In 1.23, we moved to calling `conda run <print PATH>` to determine what the PATH should be for a particular conda environment. In the dev version of reticulate we recently moved to avoiding `conda run` and instead calling `conda activate <envname>; <print PATH>`, Due to the peculiarities of conda, both approaches create a temporary script to activate an environment. In theory, going through `conda` to determine the required PATH should be more reliable than the previous approach, which had hard-coding some PATH additions. But we hadn't considered that on windows there may be restrictions on running *.bat scripts.

If you were to inspect the temporary *.bat script it would in essence contain:
```bat
CALL c:/path/to/conda.exe activate c:/path/to/conda/env
python -c "import os; print(os.environ['PATH'])"
```

---
Is there a strong motivation for using conda? Maybe a different python installation might be easier. 

You can install python via pyenv from R like this:
```R
python <- install_python("3.8.12")
virtualenv_create("r-reticulate", python)
```

Or you can install the Python binaries from www.python.org ([download link](https://www.python.org/ftp/python/3.8.10/python-3.8.10-amd64.exe))
and then run 
```R
virtualenv_create("r-reticulate", "c:/path/to/python.exe")
```

**Comment by Stephen-G** [comment]
Thanks for the suggestions! I'll need to work with IT to see if we can set this up with the permissions restrictions that are in place.

**Comment by liyifan-hub** [comment]
> 
I tried to do , but the problem is still not solved

**Comment by UppercutTKO** [comment]
It it possible with an environment variable to change the location where the .bat file is created so it does not try to run under c:\user\<user>\Appdata?

**Comment by lizelive** [comment]
packageVersion("reticulate")

[1] ‘1.26.9000’

has this error

**Comment by JosiahParry** [comment]
This issue is arising for me on Windows 10. 

Note that this issue arises for me when using escaped backslashes for the entirety of the path e.g. 

```r
reticulate::use_condaenv("C:\\path\\to\\env\\python.exe")
```

I can get around this by using forward slashes after the C: 

```r
reticulate::use_condaenv("C:\\path/to/env/python.exe")
```

```
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9              lubridate_1.8.0         here_1.0.1              lattice_0.20-45         prettyunits_1.1.1       png_0.1-7              
 [7] ps_1.7.1                assertthat_0.2.1        rprojroot_2.0.3         digest_0.6.29           utf8_1.2.2              mime_0.12              
[13] R6_2.5.1                pillar_1.8.1            rlang_1.0.6             rstudioapi_0.14         miniUI_0.1.1.1          callr_3.7.2            
[19] urlchecker_1.0.1        Matrix_1.4-1            reticulate_1.26         desc_1.4.2              devtools_2.4.5          stringr_1.4.1          
[25] htmlwidgets_1.5.4       shiny_1.7.3             compiler_4.2.1          httpuv_1.6.6            janitor_2.1.0           pkgconfig_2.0.3        
[31] pkgbuild_1.3.1          htmltools_0.5.3         tidyselect_1.2.0        tibble_3.1.8            fansi_1.0.3             crayon_1.5.2           
[37] dplyr_1.0.10            withr_2.5.0             later_1.3.0             grid_4.2.1              DBI_1.1.3               jsonlite_1.8.2         
[43] xtable_1.8-4            lifecycle_1.0.3         magrittr_2.0.3          cli_3.4.1               stringi_1.7.8           cachem_1.0.6           
[49] arcgisbinding_1.0.1.300 fs_1.5.2                promises_1.2.0.1        remotes_2.4.2           snakecase_0.11.0        ellipsis_0.3.2         
[55] vctrs_0.4.2             generics_0.1.3          tools_4.2.1             glue_1.6.2              purrr_0.3.5             processx_3.7.0         
[61] pkgload_1.3.1           fastmap_1.1.0           sessioninfo_1.2.2       memoise_2.0.1           profvis_0.3.7           usethis_2.1.6          
```

**Comment by t-kalinowski** [comment]
@JosiahParry Can you please file a new issue with the full error message and any relevant information?

**Comment by Selely91** [comment]
> Thanks for the suggestions! I'll need to work with IT to see if we can set this up with the permissions restrictions that are in place.

Hi @Stephen-G, I have the same issue. How did you solve it? Thanks

**Comment by Selely91** [comment]
I have this error too if I use reticulate 1.26 installed with conda-forge inside a conda env:
```
use_condaenv("myenv")                                 
This program is blocked by group policy. For more information, contact your system administrator.
Error in Sys.setenv(PATH = new_path) : wrong length for argument
```
